python_modules/dagster/dagster_tests/asset_defs_tests/test_assets_job.py

```diff
@@ -1941,3 +1946,23 @@ def unpartitioned_asset():
         frozenset(["hourly_asset", "unpartitioned_asset"]),
         frozenset(["daily_asset_different_start_date", "unpartitioned_asset"]),
     }
+
+
+def test_define_asset_job_source_assets():
+    @observable_source_asset
+    def foo(context):
+        return LogicalVersion("foo")
+
+    @observable_source_asset
+    def bar(context):
+        return LogicalVersion("bar")
+
+    source_asset_job = define_asset_job("source_asset_job", AssetSelection.keys("foo", "bar"))
+
+    defs = Definitions(
+        assets=[foo, bar],
+        jobs=[source_asset_job],
+    )
+
+    result = defs.get_job_def("source_asset_job").execute_in_process()
+    assert result.success
```

Comment:
Two recommendations here:

1) Ensure that the observable sources are actually executed

I use this pattern typically.

```
    executed = {}
    @observable_source_asset
    def foo(context):
        executed['foo'] = Yes
        return LogicalVersion("foo")

     ...
     
     assert executed['foo']
```

I have caught many an error this way and it provides a lot of confidence to code reviewers.

2. Demonstrate how to get the logical version out of the event stream and then assert the correct one is returned.